src/test/java/seedu/address/logic/commands/EditPropertyCommandTest.java

```diff
@@ -0,0 +1,177 @@
+
+        Property propertyInFilteredList = model.getFilteredPropertyList().get(INDEX_FIRST_PROPERTY.getZeroBased());
+        Property editedProperty = new PropertyBuilder(propertyInFilteredList)
+                .withName(VALID_NAME_BURGHLEY_DRIVE).build();
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(INDEX_FIRST_PROPERTY,
+                new EditPropertyDescriptorBuilder().withName(VALID_NAME_BURGHLEY_DRIVE).build());
+
+        String expectedMessage = String.format(EditPropertyCommand.MESSAGE_SUCCESS, editedProperty);
+
+        Model expectedModel = new ModelManager(new PropertyBook(model.getPropertyBook()), new UserPrefs());
+        expectedModel.setProperty(model.getFilteredPropertyList().get(0), editedProperty);
+
+        assertCommandSuccess(editPropertyCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_duplicatePropertyUnfilteredList_failure() {
+        Property firstProperty = model.getFilteredPropertyList().get(INDEX_FIRST_PROPERTY.getZeroBased());
+        EditPropertyDescriptor descriptor = new EditPropertyDescriptorBuilder(firstProperty).build();
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(INDEX_SECOND_PROPERTY, descriptor);
+
+        assertCommandFailure(editPropertyCommand, model, EditPropertyCommand.MESSAGE_DUPLICATE_PROPERTY);
+    }
+
+    @Test
+    public void execute_duplicatePropertyFilteredList_failure() {
+        showPropertyAtIndex(model, INDEX_FIRST_PROPERTY);
+
+        // edit Property in filtered list into a duplicate in address book
+        Property propertyInList = model.getPropertyBook().getPropertyList().get(INDEX_SECOND_PROPERTY.getZeroBased());
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(INDEX_FIRST_PROPERTY,
+                new EditPropertyDescriptorBuilder(propertyInList).build());
+
+        assertCommandFailure(editPropertyCommand, model, EditPropertyCommand.MESSAGE_DUPLICATE_PROPERTY);
+    }
+
+    @Test
+    public void execute_invalidPropertyIndexUnfilteredList_failure() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPropertyList().size() + 1);
+        EditPropertyDescriptor descriptor = new EditPropertyDescriptorBuilder()
+                .withName(VALID_NAME_BURGHLEY_DRIVE).build();
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(outOfBoundIndex, descriptor);
+
+        assertCommandFailure(editPropertyCommand, model, Messages.MESSAGE_INVALID_PROPERTY_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit filtered list where index is larger than size of filtered list,
+     * but smaller than size of address book
+     */
+    @Test
+    public void execute_invalidPropertyIndexFilteredList_failure() {
+        showPropertyAtIndex(model, INDEX_FIRST_PROPERTY);
+        Index outOfBoundIndex = INDEX_SECOND_PROPERTY;
+        // ensures that outOfBoundIndex is still in bounds of address book list
```

Comment:
Remember to update the javadocs here too

Reply:
shld be fixed!